lib/rules/complexity.js

```diff
@@ -11,8 +11,15 @@
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
+    var option = context.options[0],
+        THRESHOLD = 20;
 
-    var THRESHOLD = (typeof context.options[0] !== "undefined") ? context.options[0] : 20;
+    if (option.hasOwnProperty("maximum") && typeof option.maximum === "number") {
+        THRESHOLD = option.maximum;
+    }
+    if (typeof option === "number") {
+        THRESHOLD = option;
+    }
```

Comment:
It appears that the pattern established in the project is to keep the logic for a rule contained within the rule file. But, I don't have enough experience with the project yet, so I'll have to refer to one of the maintainers on this.